src/stores/room-list/filters/NameFilterCondition.ts

```diff
@@ -56,6 +56,14 @@ export class NameFilterCondition extends EventEmitter implements IFilterConditio
                 return true;
             }
         }
-        return room.name && removeHiddenChars(room.name).toLowerCase().includes(lcFilter);
+
+        if (!room.name) return false; // should realisitically not happen: the js-sdk always calculates a name
+
+        // Note: we have to match the filter with the removeHiddenChars() room name because the
+        // function strips spaces and other characters (M becomes RN for example, in lowercase).
+        // We also doubly convert to lowercase to work around oddities of the library.
```

Comment:
this feels wrong, can you give an example? I would think converting *before* is most correct

Reply:
There are some languages out there which have characters that don't get squished with `.toLowerCase()` and thus get converted to uppercase representations on our end. I don't have specific examples, and would rather avoid digging it out.

This is actually what the old room list does, it just doesn't document it.